profile/src/main/java/com/scottlogic/deg/profile/reader/ConstraintReaderHelpers.java

```diff
@@ -44,18 +44,15 @@ public static Object getValidatedValue(ConstraintDTO dto) {
     public static Set<Object> getValidatedValues(ConstraintDTO dto) {
         if (dto.values == null) {
             throw new InvalidProfileException(String.format(
-                "Field [%s]: Couldn't recognise 'values' property, it must not contain 'null'",
-                dto.field
+                "Field [%s]: Requires a 'values' property but instead a 'value' property has been added.",
+                    dto.field
             ));
```

Comment:
Here, the error message has been hard-coded, whereas what is really wanted is a variable message depending on the mistake the user made.

For example, `{ "field": "Column 1", "is": "inSet", "value": ["a", "b"] }` should give an error message like "Requires a 'values' property but instead a 'value' property has been added.".

And, `{ "field": "Column 1", "is": "inSet", "invalidProperty1": ["a", "b"], "invalidProperty2": ["a", "b"] }` should give an error message like "Requires a 'values' property but instead the properties "invalidProperty1" and "invalidProperty2" were added".

This could possibly be done by adding a helper method to the `ConstraintDTO` class. 

Sorry if that wasn't clear on the issue!